doc/configuration.mld

```diff
@@ -0,0 +1,35 @@
+{0 How OCamlFormat computes its configuration}
+
+{1 Configuration files}
+
+Ocamlformat fetches the following files on the FS:
+
++ [.git], [.hg] or [dune-project]
++ [.ocamlformat] and [.ocp-indent] files
++ [.ocamlformat-ignore] files
++ [.ocamlformat-enable] files
```

Comment:
Numbers are meaningful and should be written down. Also merging 3 and 4 that causes repeating text later.
```suggestion
1. [.git], [.hg] or [dune-project]
2. [.ocamlformat] and [.ocp-indent]
3. [.ocamlformat-ignore] and [.ocamlformat-enable]
```